File: src/ch03/code_4.py
from typing import List

from rich import print


def search_min_pair(nums1: List[int], nums2: List[int], threshold: int) -> int:
    min_val: int = 1e10
    for num1 in nums1:
        for num2 in nums2:
            pair: int = num1 + num2
            if pair < threshold:
                continue
            if pair < min_val:
                min_val = pair

    return min_val


if __name__ == '__main__':
    nums1: List[int] = [8, 5, 4]
    nums2: List[int] = [4, 1, 9]
    threshold: int = 10
    result: int = search_min_pair(nums1=nums1, nums2=nums2, threshold=threshold)
    
    print(result)

File: src/ch03/code_4.rs
fn search_min_pair(nums1: Vec<i32>, nums2: Vec<i32>, threshold: i32) -> i32 {
    let mut min_val: i32 = 20000000;
    for num1 in &nums1 {
        for num2 in &nums2 {
            let pair: i32 = *num1 + *num2;
            if pair < threshold {
                continue;
            }
            if pair < min_val {
                min_val = pair;
            }
        }
    }
    min_val
}

fn main() {
    let nums1: Vec<i32> = vec![8, 5, 4];
    let nums2: Vec<i32> = vec![4, 1, 9];
    let threshold: i32 = 10;
    let result: i32 = search_min_pair(nums1, nums2, threshold);

    println!("{}", result);
}